provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -480,6 +486,39 @@ func (c *Client) addNetworkDevice(
 	return &networkDevice, nil
 }
 
+// generateMAC generates a random hardware address that meets VMWare
+// requirements for MAC address: 00:50:56:XX:YY:ZZ where XX is between 00 and 3f.
```

Comment:
can we link to the docs you used for this information? That way we can continue to make sure its accurate.